File: dao/src/main/java/com/alta_v2/game/dao/aop/executionTime/PrintExecutionTime.java
package com.alta_v2.game.dao.aop.executionTime;

import com.google.inject.BindingAnnotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Provides the annotation to mark method that for which execution time should be logged.
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@BindingAnnotation
public @interface PrintExecutionTime {

    String operationName() default "";

}

File: shared/src/main/java/com/alta_v2/aop/executionTime/PrintExecutionTime.kt
package com.alta_v2.aop.executionTime

import com.google.inject.BindingAnnotation

/**
 * Provides the annotation to mark method that for which execution time should be logged.
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@BindingAnnotation
annotation class PrintExecutionTime(val operationName: String = "")